File: redsheet-portal/src/app/database-models/project-carry-forward.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule,  vProjectCarryForwardRecord  } from './'
export class ProjectCarryForward {
    constructor(projectCarryForward?: ProjectCarryForward  | vProjectCarryForwardRecord )  {
        let properties = [ "carryForwardValue", "projectCarryForwardId", "projectModuleId",];
        for (let property of properties)
            if (projectCarryForward && projectCarryForward[property])
                this[property] = projectCarryForward[property];
    }
 
    @required()
    @maxLength(350)
	carryForwardValue : string =   undefined;
 
	projectCarryForwardId : number =   0 ;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;


}

File: RedSheet.DbEntities/Models/ProjectCarryForward.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectCarryForwards", Schema = "dbo")]
	[Serializable]
	public partial class ProjectCarryForward  
	{
 
	[Required] 
	[MaxLength(400)]
	public string CarryForwardValue { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectCarryForwardId { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 

	public ProjectCarryForward()
	{
    }

    }
}

